openapi.go

```diff
@@ -631,28 +632,52 @@ func (p *Path) ProtoEndpoints(annotate bool, base, path string) string {
 func (p *Path) ProtoMessages(path string, defs map[string]*Items) string {
 	var out bytes.Buffer
 	if p.Get != nil {
-		endpointName := PathMethodToName(path, "get")
+		var endpointName string
+		if p.Get.OperationID == "" {
+			endpointName = PathMethodToName(path, "get")
+		} else {
+			endpointName = p.Get.OperationID
+		}
+
 		msg := p.Get.protoMessages(p.Parameters, endpointName, defs)
 		if msg != "" {
 			out.WriteString(msg)
 		}
 	}
 	if p.Put != nil {
-		endpointName := PathMethodToName(path, "put")
+		var endpointName string
+		if p.Put.OperationID == "" {
+			endpointName = PathMethodToName(path, "get")
+		} else {
+			endpointName = p.Put.OperationID
+		}
+
 		msg := p.Put.protoMessages(p.Parameters, endpointName, defs)
 		if msg != "" {
 			out.WriteString(msg)
 		}
 	}
 	if p.Post != nil {
-		endpointName := PathMethodToName(path, "post")
+		var endpointName string
+		if p.Post.OperationID == "" {
+			endpointName = PathMethodToName(path, "get")
+		} else {
+			endpointName = p.Post.OperationID
+		}
+
 		msg := p.Post.protoMessages(p.Parameters, endpointName, defs)
 		if msg != "" {
 			out.WriteString(msg)
 		}
 	}
 	if p.Delete != nil {
-		endpointName := PathMethodToName(path, "delete")
+		var endpointName string
+		if p.Delete.OperationID == "" {
```

Comment:
Instead of all this repetition, could we possibly move this 'if' statement into `PathMethodToName`? The func may require a param change (and maybe a rename?), but it would probably DRY things up a tad.

Reply:
Yep, will be done

Reply:
Done